yamale/command_line.py

```diff
@@ -85,18 +77,28 @@ def _validate_dir(root, schema_name, cpus, parser, strict):
 
     print('Found %s yaml files.' % len(res))
     print('Validating...')
+    results = []
```

Comment:
Instead of collating and returning results, you could raise an exception with all the errors. Then the `_router` method doesn't have to change and the exception would get passed directly to `main` where it can be handled.